plugins/LayerView/LayerView.py

```diff
@@ -392,10 +389,13 @@ def _updateWithPreferences(self):
 
         self.setLayerViewType(int(float(Preferences.getInstance().getValue("layerview/layer_view_type"))));
 
-        self.setExtruderOpacity(0, float(Preferences.getInstance().getValue("layerview/extruder0_opacity")))
-        self.setExtruderOpacity(1, float(Preferences.getInstance().getValue("layerview/extruder1_opacity")))
-        self.setExtruderOpacity(2, float(Preferences.getInstance().getValue("layerview/extruder2_opacity")))
-        self.setExtruderOpacity(3, float(Preferences.getInstance().getValue("layerview/extruder3_opacity")))
+        extruder_nr = 0
+        for extruder_opacity in Preferences.getInstance().getValue("layerview/extruder_opacities").split(","):
+            try:
+                opacity = float(extruder_opacity)
+            except ValueError:
+                opacity = 1.0
+            self.setExtruderOpacity(extruder_nr, opacity)
```

Comment:
So you always set the extruderOpacity of the first extruder? I'm guessing you missed a extruder_nr += 1 somehwere in the loop.